### ISSUE TITLE
Value Checker unsafe cast warning for primitive wrappers applied to constants

### ISSUE THREAD

**Issue by gokaco** [typetools/checker-framework#2353]
Warnings are occurring while running the given piece of code:-
command-
`javac -processor index ex2.java`
code-
```
public class Issue2353 {

    public static void main(String args[]) {
        Integer a = new Integer(1);
        Integer b = new Integer("2");
    }
}
```

Output:
```
Issue2353.java:5: warning: [cast.unsafe.constructor.invocation] constructor invocation cast from "@UnknownVal Integer" to "@IntVal(1) Integer" cannot be statically verified
        Integer b = new Integer(1);
                    ^
Issue2353.java:6: warning: [cast.unsafe.constructor.invocation] constructor invocation cast from "@UnknownVal Integer" to "@IntVal(2) Integer" cannot be statically verified
        Integer c = new Integer("2");
                    ^
2 warnings
```
How to make it verify statically?

**Comment by mernst** [comment]
Thanks for the report.  This is definitely a bug caused by 19807da029d805318677f0a18445452554bd8b73.  `BaseTypeVisitor` should not issue the warning for annotations that are soundly introduced by the type system.

PS: A tip to improve the formatting of GitHub issues you report, surround multi-line code or output with ```...```.  (I have done that for this issue.)
